components/status/HeatMapColorBar.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div>
+    <div id="parent-container">
+      <div
+        id="startup-heatmap"
+        class="div__heatmap-gradient-holder"
+        :style="value"
+      ></div>
+    </div>
+    <!-- prettier-ignore -->
+    <span class="span__heatmap-scale-higher-value">{{ upper_range }} {{ units }}</span>
+    <!-- prettier-ignore -->
+    <span class="span__heatmap-scale-lower-value" :style="top_shift">{{ lower_range }} {{ units }}</span
+    >
+  </div>
+</template>
+<script>
+/**
+ **/
+export default {
+  name: "HeatMapColorBar",
+  props: {
+    gradient_uuid: {
+      type: String,
+      default: "",
+    },
+    lower_range: {
+      type: Number,
+      required: true,
+    },
+    upper_range: {
+      type: Number,
+      required: true,
+    },
+    gradient_range: {
+      type: Array,
+      default: function () {
+        return [];
+      },
+    },
+    heatmap_height: {
+      type: Number,
+      required: true,
+    },
+    units: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      heatmap_uuid: "0f81155d-23ec-4790-a3fe-92a6cc7c3c47",
+      uuid_defined_colormap: [
+        { color: "#2c7bb6", offset: "0%" },
+        { color: "#00a6ca", offset: "12.5%" },
+        { color: "#00ccbc", offset: "25%" },
+        { color: "#90eb9d", offset: "37.5%" },
+        { color: "#ffff8c", offset: "50%" },
+        { color: "#f9d057", offset: "62.5%" },
+        { color: "#f29e2e", offset: "75%" },
+        { color: "#e76818", offset: "87.5%" },
+        { color: "#d7191c", offset: "100%" },
+      ],
+    };
+  },
+  computed: {
+    top_shift() {
+      return "top: " + (this.heatmap_height - 16).toString() + "px;";
+    },
+  },
+  watch: {
+    gradient_uuid: function () {
+      this.common_api();
+    },
+    gradient_range: function () {
+      this.common_api();
+    },
+    heatmap_height: function () {
+      this.common_api();
+    },
+  },
+  created() {
+    this.value = "";
```

Comment:
we talked this morning about how `value` is not a sufficiently descriptive variable name.  this variable relates to the color gradient, so the variable name should reflect that.

Reply:
'value' ==> 'gradient_cssproperty'